### DISCUSSION TITLE
Authorization Code Flow Example — Eg. Shopify Public App

### DISCUSSION THREAD

**Discussion by avcohen** [nextauthjs/next-auth#935] [Help]
**Your question**
I'm having issue getting this linked up with Shopify as a custom provider. Does anyone have any examples using Shopify or an authorization code flow with custom provider like Shopify?

**What are you trying to do**
Setup Next-Auth with a public Shopify App.

**Feedback**
*Documentation refers to searching through [online documentation](https://next-auth.js.org), code comments and issue history. The example project refers to [next-auth-example](https://github.com/iaincollins/next-auth-example).*

* [x] Found the documentation helpful
* [x] Found documentation but was incomplete
* [x] Could not find relevant documentation
* [x] Found the example project helpful
* [ ] Did not find the example project helpful

**Comment by cjgaliana** [comment]
Are there any news about this?

**Comment by StefanSelfTaught** [comment]
Something like this won't work?
```js
providers: [
    {
      id: 'shopify',
      name: 'Shopify',
      type: 'oauth',
      version: '2.0',
      scope: "write_orders,read_customers",
      params: { grant_type: 'authorization_code' },
      accessTokenUrl: 'https://{yourShop}.myshopify.com/admin/oauth/access_token',
      authorizationUrl: 'https://{yourShop}.myshopify.com/admin/oauth/authorize',
      profileUrl: 'https://{yourShop}.myshopify.com/admin/api/2021-04/users/current.json',
      async profile(profile) {
        return {
          id: profile.id,
          name: `${profile.first_name} ${profile.last_name}`,
          email: profile.email
        }
      },
      clientId: process.env.SHOPIFY_CLIENT_ID,
      clientSecret: process.env.SHOPIFY_CLIENT_SECRET,
    },
]

**Reply by cmnstmntmn** [reply]
hei, 
i'm having an `401` oauth_callback_error , even if the app is installed. 

```json
{"errors":"[API] Invalid API key or access token (unrecognized login or wrong password)"}
```

do i need to pass any kind of headers?
thanks

**Reply by harishannam** [reply]
I am getting the same `401` error. @cmnstmntmn did you find a solution by any chance?

**Comment by jsullivan5** [comment]
Did anyone ever get this working?  I'm having trouble getting the pre and post install checks against HMAC, nonce etc. and keep getting oauth callback and TLS check errors.

**Reply by michpenn** [reply]
I'm also running into the same issues. Definitely curious if anybody was able to get this working with Shopify. If not, I'm curious what alternatives people used.

**Reply by driespieters** [reply]
Same!

**Reply by denik1981** [reply]
I'm on the same track. This shouldn't be marked as answered.

**Reply by Sahejmaharjan1** [reply]
any updates on this?

**Comment by denik1981** [comment]
Is this possible or not? Seems to me Shopify has overcomplicated the way it should be handleling an OAuth authentication.

**Comment by mgavin-origen** [comment]
another with the same problem
this is not answered
